devicetypes/smartthings/centralite-dimmer.src/centralite-dimmer.groovy

```diff
@@ -54,103 +43,54 @@ metadata {
-			value = Integer.parseInt(reportValue, 16) / 10
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		log.info event
+		if (event.name == "power") {
+			if (device.getDataValue("manufacturer") != "OSRAM") {       //OSRAM devices do not reliably update power
+				event.value = (event.value as Integer) / 10             //TODO: The divisor value needs to be set as part of configuration
+				sendEvent(event)
+			}
+		}
+		else {
+			sendEvent(event)
 		}
-	} else if (description?.startsWith("on/off:")) {
-		log.debug "Switch command"
-		name = "switch"
-		value = description?.endsWith(" 1") ? "on" : "off"
 	}
-
-	def result = createEvent(name: name, value: value)
-	log.debug "Parse returned ${result?.descriptionText}"
-	return result
-}
-
-def parseDescriptionAsMap(description) {
-	(description - "read attr - ").split(",").inject([:]) { map, param ->
-		def nameAndValue = param.split(":")
-		map += [(nameAndValue[0].trim()):nameAndValue[1].trim()]
+	else {
+		log.warn "DID NOT PARSE MESSAGE for description : $description"
+		log.debug zigbee.parseDescriptionAsMap(description)
 	}
 }
 
-// Commands to device
-def on() {
-	[
-			'zcl on-off on',
-			'delay 200',
-			"send 0x${zigbee.deviceNetworkId} 0x01 0x${zigbee.endpointId}",
-			'delay 500'
-	]
-}
-
 def off() {
-	[
-			'zcl on-off off',
-			'delay 200',
-			"send 0x${zigbee.deviceNetworkId} 0x01 0x${zigbee.endpointId}",
-			'delay 500'
-	]
+	zigbee.off()
 }
 
-def setLevel(value) {
-	log.trace "setLevel($value)"
-	sendEvent(name: "level", value: value)
-    def level = hexString(Math.round(value * 255/100))
-	def cmd = "st cmd 0x${device.deviceNetworkId} 1 8 4 {${level} 2000}"
-	log.debug cmd
-	cmd
+def on() {
+	zigbee.on()
 }
 
-def meter() {
-	"st rattr 0x${device.deviceNetworkId} 1 0xB04 0x50B"
+def setLevel(value) {
+	zigbee.setLevel(value)
 }
 
 def refresh() {
-	"st rattr 0x${device.deviceNetworkId} 1 0xB04 0x50B"
+	zigbee.onOffRefresh() + zigbee.levelRefresh() + zigbee.simpleMeteringPowerRefresh() + zigbee.electricMeasurementPowerRefresh() + zigbee.onOffConfig() + zigbee.levelConfig() + zigbee.simpleMeteringPowerConfig() + zigbee.electricMeasurementPowerConfig()
 }
 
 def configure() {
-	[
-		"zdo bind 0x${device.deviceNetworkId} 1 1 8 {${device.zigbeeId}} {}", "delay 200",
-		"zdo bind 0x${device.deviceNetworkId} 1 1 6 {${device.zigbeeId}} {}", "delay 200",
-		"zdo bind 0x${device.deviceNetworkId} 1 1 0xB04 {${device.zigbeeId}} {}"
-	]
-}
-
-private hex(value, width=2) {
-    def s = new BigInteger(Math.round(value).toString()).toString(16)
-    while (s.size() < width) {
-        s = "0" + s
-    }
-    s
+	log.debug "Configuring Reporting and Bindings."
+	zigbee.onOffConfig() + zigbee.levelConfig() + zigbee.simpleMeteringPowerConfig() + zigbee.electricMeasurementPowerConfig() + zigbee.onOffRefresh() + zigbee.levelRefresh() + zigbee.simpleMeteringPowerRefresh() + zigbee.electricMeasurementPowerRefresh()
```

Comment:
The production version of zigbee-dimmer-power just calls `refresh()` here. Did you use an earlier version?